components/clp-package-utils/clp_package_utils/scripts/native/compress.py

```
@@ -153,41 +157,152 @@ def _generate_clp_io_config(
+    :raises ValueError: If validation fails
+    """
+    if len(urls) == 0:
+        raise ValueError("No URLs provided")
+
+    parsed_urls = []
+    for url in urls:
+        try:
+            region_code, bucket, key = parse_s3_url(url)
+            parsed_urls.append((region_code, bucket, key))
+        except ValueError as e:
+            raise ValueError(f"Failed to parse URL '{url}': {e}")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)